[Profiling] Replacing TopN functions table for EuiGrid

cauemarcondes opened the pull request:
**TopN functions**
<img width="1795" alt="Screenshot 2023-07-04 at 12 01 30 PM" src="https://github.com/elastic/kibana/assets/55978943/9acab124-ffb5-4c5f-8e53-fdfe2bd7a247">

**Diff TopN functions**
<img width="1799" alt="Screenshot 2023-07-04 at 12 03 04 PM" src="https://github.com/elastic/kibana/assets/55978943/ec0acf68-7337-48ab-a031-49420a2d770a">

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)

</p>
</details>

jbcrail reviewed (commented):
I did a first pass to review the code and it looks good. I will follow up on Wednesday to try it out with live data.

jbcrail commented:
Here are a few miscellaneous items that I noticed after setting up a cloud instance with live data:

1. This is a very general observation with no concrete data, but it seemed that the grid rendered slower than the prior implementation. Perhaps this is a function of running Kibana locally and pulling the data from the cloud. Or maybe it's inherent to the EuiGrid. I'm not exactly sure without further investigation.

2. I saw at least one React warning in the console:

```
Warning: Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state.
```

3. For non-differential TopN functions, the data in the function row does not match with the data in the frame information window. For instance, see the values for total CPU, self CPU, and both samples. In the following screenshots, I clicked on the row for rank 21.

**Before**

![image](https://github.com/elastic/kibana/assets/6038/390ecb01-358c-47a6-b8fd-a2e70cf760b5)

**After**

![image](https://github.com/elastic/kibana/assets/6038/b20cf016-c2a7-4bf1-b84e-24f354e1ee25)

4. When the button for fullscreen mode is clicked and then closed, the grid does not return to the original size (fit to the available width). You can see the grid overflow past the right side of the page in the second screenshot. I'm not sure if this is a EuiGrid bug or controllable on our end.

**Before**

![image](https://github.com/elastic/kibana/assets/6038/82953542-c399-4e20-9759-4d8e875d796e)

**After**

![image](https://github.com/elastic/kibana/assets/6038/398e06e4-0282-4680-a103-c14d8d6ec0b3)

5. Would it be clearer from a UI perspective nd/or easier for internalization to swap the abbreviation in the `Samples` column (`estd.`) with the same question icon used in the CPU columns?

6. I noticed that the "Annualized CO2" column is sorted using A-Z vs Z-A, not High-Low vs Low-High. Since the annualized CO2 is ultimately a numeric value but is represented by a string, this creates an odd sorted list that may not match a user's expectation.

cauemarcondes commented:
> Would it be clearer from a UI perspective nd/or easier for internalization to swap the abbreviation in the Samples column (estd.) with the same question icon used in the CPU columns?

Looks good to me. I added an `Estimated values` in the hint.
cc: @iogbole

cauemarcondes commented:
>6. I noticed that the "Annualized CO2" column is sorted using A-Z vs Z-A, not High-Low vs Low-High. Since the annualized CO2 is ultimately a numeric value but is represented by a string, this creates an odd sorted list that may not match a user's expectation.

Thanks, I fixed it. I also noticed that the `Samples` column wasn't being sorted correctly. I also fixed it.

jbcrail commented:
Thanks for fixing those items. I retested and only saw one small issue with two consecutive tildes in the frame information window, specifically for the `CO2 emission` and `Dollar cost` estimates:

![image](https://github.com/elastic/kibana/assets/6038/43d61746-7dc8-4995-9ebc-9c5b337461e6)

jbcrail reviewed (approved):
Looks good :+1: :rocket:
